verilog/analysis/checkers/token_stream_lint_rule.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/token_stream_lint_rule.h"
+
+#include <set>
+#include <string>
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "common/analysis/citation.h"
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/bound_symbol_manager.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::GetStyleGuideCitation;
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::SyntaxTreeContext;
+using verible::matcher::Matcher;
+
+// Register TokenStreamLintRule
+VERILOG_REGISTER_LINT_RULE(TokenStreamLintRule);
+
+absl::string_view TokenStreamLintRule::Name() { return "forbid-line-continuations"; }
+const char TokenStreamLintRule::kTopic[] = "forbid-line-continuations";
+const char TokenStreamLintRule::kMessage[] =
+    "The lines can't be continued with \'\\\', use C-style multiline "
```

Comment:
@hzeller should we call this `C-style concatenation` even though it isn't exactly the same? Shouldn't we just recommend to use the `concatenation operator` or explicitly say to concatenate using braces?

Reply:
I agree, I think `concatenation operator with braces` would be easier to understand in this context.